### PR TITLE
Skip mixed in ParamTypeByMethodCallTypeRector as not specific

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: src/Application/FileProcessor.php [hunk 1]
```diff
@@ -150,13 +150,29 @@ private function printFile(File $file, Configuration $configuration, string $fil
...
+            $originalFileContent = $file->getOriginalFileContent();
+            $ltrimOriginalFileContent = ltrim($originalFileContent);
+
+            // handle space before <?php
+            $ltrimNewContent = Strings::replace($newContent, self::OPEN_TAG_SPACED_REGEX, '<?php');
+            $ltrimOriginalFileContent = Strings::replace(
+                $ltrimOriginalFileContent,
+                self::OPEN_TAG_SPACED_REGEX,
+                '<?php'
+            );
```
Comment (samsonasik) [inline src/Application/FileProcessor.php]: it seems merge conflict? this unrelated change and regex change should be checked after ltrim(original content) === newContent compared

Comment (samsonasik) [reply src/Application/FileProcessor.php]: For note: php-parser v4 use ltrim on print format preserving:

https://github.com/nikic/PHP-Parser/blob/a6303e50c90c355c7eeee2c4a8b27fe8dc8fef1d/lib/PhpParser/PrettyPrinterAbstract.php#L503

while v5 is not:

https://github.com/nikic/PHP-Parser/blob/f5adbb5e3fa57d2ff7b9ea179b88bcd2ced153e0/lib/PhpParser/PrettyPrinterAbstract.php#L553

so the original check verify `ltrim()` on very first level, no need regex check, see:

- https://github.com/rectorphp/rector/issues/8228

on php-parser v5, very first `ltrim()` no longer needed,

Then on next lines, it should verify deep open tag only, so regex is used, that's why `OPEN_TAG_SPACED_REGEX` is used.

see PR:

- https://github.com/rectorphp/rector-src/pull/5701

for it.

This verify only on **no diff change**, as compare original content vs printed content. That will be re-verify on php-parser v5.

Comment (TomasVotruba) [reply src/Application/FileProcessor.php]: Only bad rebase :)
